cmd/kyma/test/logs/cmd.go

```diff
@@ -0,0 +1,143 @@
+package logs
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/kyma-project/cli/internal/logs"
+	"strings"
+
+	oct "github.com/kyma-incubator/octopus/pkg/apis/testing/v1alpha1"
+	"github.com/kyma-project/cli/cmd/kyma/test"
+	"github.com/kyma-project/cli/internal/cli"
+	"github.com/kyma-project/cli/internal/kube"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	defaultIgnoredContainers = []string{"istio-init", "istio-proxy", "manager"}
+	defaultLogsInStatus      = string(oct.TestFailed)
+)
+
+type command struct {
+	opts *options
+	cli.Command
+}
+
+func NewCmd(o *options) *cobra.Command {
+	cmd := command{
+		Command: cli.Command{Options: o.Options},
+		opts:    o,
+	}
+
+	cobraCmd := &cobra.Command{
+		Use:   "logs <test-suite-1> <test-suite-2> ... <test-suite-N>",
+		Short: "Shows the logs of a related test executions pods.",
+		Long: `Use this command to display the logs of a test executing for given test suite.
+
+To print the status of specific test cases, ` + "`run kyma test logs testSuiteOne testSuiteTwo`" + `.
+Provide at least one test suite name.
+`,
+
+		RunE: func(_ *cobra.Command, args []string) error { return cmd.Run(args) },
+	}
+
+	cobraCmd.Flags().Bool("help", false, "Displays help for the command.")
+	cobraCmd.Flags().StringVar(&o.InStatus, "in-status", defaultLogsInStatus, "Display logs only from testing pods in given status.")
+	cobraCmd.Flags().StringSliceVar(&o.IngoredContainers, "ignored-containers", defaultIgnoredContainers, "Specifies the containers name which are ignored when fetching logs from testing pods. Takes comma-separated list.")
+
+	return cobraCmd
+}
+
+func (cmd *command) Run(args []string) error {
+	if err := cmd.validateFlags(); err != nil {
+		return err
+	}
+
+	if len(args) < 1 {
+		return fmt.Errorf("Test suite name required")
+	}
+
+	k8sClient, err := kube.NewFromConfig("", cmd.KubeconfigPath)
```

Comment:
```suggestion
        var err error
	cmd.K8s, err = kube.NewFromConfig("", cmd.KubeconfigPath)
```

Reply:
I will apply your suggestion but two cents from my side:

I'm not convinced of that approach because

> When returning an error from a function with multiple return values, idiomatic Go code also will set each non-error value to a zero value

and basically it also should be followed here. IMO assignment to cmd.K8s should be done only if there is no error. In case of error, you may have sth in cmd.K8s. I know that in that given code is easy to say that it will not cause other problems but basically when I'm writing code then I'm doing that always in the same way

Reply:
Well, based on that quote this applies about perfectly. Since `cmd.K8s` is an interface, its zero value is nil. So if `kube.NewFromConfig()` follows that, it will return nil for the first return value if there is an error.
So with that principle we can safely assign it.